File: python/sieve/sieve.py
def sieve(max):
  table = dict.fromkeys(range(2, max+1))

  for current_num in range(2, max + 1):
    if not table[current_num] is None:
      continue
    table[current_num] = True
    for m in range(current_num * 2, max+1, current_num):
        table[m] = False

  return [key for key in table.keys() if table[key]]

File: csharp/sieve/Sieve.cs
using System.Collections.Generic;
using System.Linq;

public class Sieve
{
    public static int[] Primes(int max)
    {
        var map = new Dictionary<int, bool>();
        for (var i = 2; i <= max; i++)
        {
            if (map.ContainsKey(i)) continue;
            map[i] = true;
            for (var j = i * i; j <= max; j += i)
            {
                map[j] = false;
            }
        }
        return map.Where(x => x.Value).Select(x => x.Key).OrderBy(x => x).ToArray();
    }
}